From 76b2b2942e15f1b168163b1eb8064fa906b4d149 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 5 Feb 2019 01:26:01 +0100
Subject: [PATCH 1/1] tpm: simplify: tpm_set_global_lock()

When in pack_byte_string() memcpy() is called for size 0 the source buffer
address has no relevance. So we can use NULL here.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/tpm-v1.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/lib/tpm-v1.c b/lib/tpm-v1.c
index 3e89f84544..b4498e6afc 100644
--- a/lib/tpm-v1.c
+++ b/lib/tpm-v1.c
@@ -211,9 +211,7 @@ u32 tpm_nv_write_value(struct udevice *dev, u32 index, const void *data,
 
 uint32_t tpm_set_global_lock(struct udevice *dev)
 {
-	u32 x;
-
-	return tpm_nv_write_value(dev, TPM_NV_INDEX_0, (uint8_t *)&x, 0);
+	return tpm_nv_write_value(dev, TPM_NV_INDEX_0, NULL, 0);
 }
 
 u32 tpm_extend(struct udevice *dev, u32 index, const void *in_digest,
-- 
2.20.1

